qualtran/surface_code/data_block.py

```diff
@@ -76,3 +83,93 @@ def data_error(self, n_algo_qubits: int, n_cycles: int, phys_err: float) -> floa
         return data_cells * self.qec_scheme.logical_error_rate(
             physical_error_rate=phys_err, code_distance=self.data_d
         )
+
+    def n_cycles_to_consume_a_magic_state(self) -> int:
+        return self.data_d
+
+
+@frozen
+class CompactDataBlock(SimpleDataBlock):
+    r"""The compact data block uses a fixed code distance and routing overhead.
+
+    The compact data block lays $n$ qubit batches in grid of shape (3, $n/2$) where
+    the data batches are lined in the first and last row with the middle row being
+    an ancilla region. This lowers the memory footprint of the block at the cost of an
+    increased number of cycles to consume a magic state.
+
+    References:
+        [A Game of Surface Codes](https://arxiv.org/abs/1808.02892)
+        page 7, figure 9
+    """
+
+    routing_overhead: float = 0.5
```

Comment:
Document all the args in each of the concrete subclasses to aid the user; and you can describe why routing overhead is 0.5

When constructing a `CompactDataBlock`, you can specify a non-default `routing_overhead`. is this desired?

Reply:
the routing overhead for the compact class is different from the others. the others have routing overhead that is just a number * n_qubits, but for the compac class it's `n + sqrt(8n)`

Reply:
I will add a validator for that

Reply:
seems weird tho -- it should be a property that doesn't show up at all in the constructor. I know it's because of the interaction between attrs and inheritance but there's probably a more elegant way

Reply:
I modified my comment, I will use `field(init=False, default=42)` instead

Reply:
this way it won't show up in init